packages/core/src/asciidoctor-core-api.js

```diff
@@ -1096,6 +1096,14 @@ AbstractNode.prototype.getNodeName = function () {
   return this.node_name
 }
 
+/**
+ * Sets the name of the node.
+ *
+ * @param {string} nodeName - The name to be set for the node.
+ */
+AbstractNode.prototype.setNodeName = function (nodeName) {
+  this.node_name = nodeName
+}
```

Comment:
`node_name` is declared as an attribute reader: https://github.com/asciidoctor/asciidoctor/blob/7d856042fdd278f87d4c480180440e519fa1e97c/lib/asciidoctor/abstract_node.rb#L23-L24

@mojavelinux should we use `attr_accessor` on `node_name`?

Reply:
I'm going to warn you, we are really getting into some internal state here. The node_name variable has never been formalized and allowing it to be changed is really undefined. If anything, setting the context should set the node_name so it remains internal. But for an inline node, the node_name is "inline_" + the context.

Reply:
I'm reluctant to add it to the "porcelain" API. I would be in favour of adding two fields in the TypeScript definition `node_name` and `context`.

That way `block.context = block.node_name = newName` won't raise a warning.

Reply:
Mmm, yes. 🤔

It sounds like I'm skating very close to breaking stuff, now. 

I have a vague memory of having to set the node name in the JavaScript API, but that wasn't needed for the Ruby API. I'll try a bit of test code to see if I can do this without fiddling with the node name. 

Or perhaps I need to look at doing this a different way. Rather than alter the context, it would be safer to remove olist block and replace it with a colist built from scratch. (Would be more work though).

Reply:
> I'm reluctant to add it to the "porcelain" API. I would be in favour of adding two fields in the TypeScript definition `node_name` and `context`.
> 
> 
> 
> That way `block.context = block.node_name = newName` won't raise a warning.

They would be picked up by the underlying Ruby API if I just add two fields in the TypeScript file?

Reply:
Yes, I think so.
As far as I recall, Opal is using un-prefixed fields for `attr_reader`.

Reply:
Yup, that seems to work 👍🏾